qiskit_nature/problems/second_quantization/lattice/lattice_model_problem.py

```diff
@@ -0,0 +1,125 @@
+from .models.lattice_model import LatticeModel
+from ..base_problem import BaseProblem
+
+
+class LatticeModelProblem(BaseProblem):
+    """Lattice Model Problem class to create second quantized operators from a lattice model."""
+
+    def __init__(self, lattice_model: LatticeModel) -> None:
+        """
+        Args:
+            lattice_model: A lattice model class to create second quantized operators.
+        """
+        super().__init__()
+        self._lattice_model = lattice_model
+        self._main_property_name = "LatticeEnergy"
+
+    def second_q_ops(self) -> ListOrDictType[SecondQuantizedOp]:
+        """Returns the second quantized operators created based on the lattice models.
+
+        Returns:
+            A ``list`` or ``dict`` of
+            :class:`~qiskit_nature.operators.second_quantization.SecondQuantizedOp`
+        """
+        second_q_ops: ListOrDictType[SecondQuantizedOp] = self._lattice_model.second_q_ops()
+        if settings.dict_aux_operators:
+            second_q_ops = {self._main_property_name: second_q_ops}
+        else:
+            second_q_ops = [second_q_ops]
+
+        return second_q_ops
+
+    def interpret(
+        self,
+        raw_result: Union[EigenstateResult, EigensolverResult, MinimumEigensolverResult],
+    ) -> LatticeModelResult:
+        """Interprets a raw result in the context of this transformation.
+
+        Args:
+            raw_result: a raw result to be interpreted
+
+        Returns:
+            A lattice model result.
+        """
+        eigenstate_result = None
+        if isinstance(raw_result, EigenstateResult):
+            eigenstate_result = raw_result
+        elif isinstance(raw_result, EigensolverResult):
+            eigenstate_result = EigenstateResult()
+            eigenstate_result.raw_result = raw_result
+            eigenstate_result.eigenenergies = raw_result.eigenvalues
+            eigenstate_result.eigenstates = raw_result.eigenstates
+            eigenstate_result.aux_operator_eigenvalues = raw_result.aux_operator_eigenvalues
+        elif isinstance(raw_result, MinimumEigensolverResult):
+            eigenstate_result = EigenstateResult()
+            eigenstate_result.raw_result = raw_result
+            eigenstate_result.eigenenergies = np.asarray([raw_result.eigenvalue])
+            eigenstate_result.eigenstates = [raw_result.eigenstate]
+            eigenstate_result.aux_operator_eigenvalues = [raw_result.aux_operator_eigenvalues]
+        result = LatticeModelResult()
+        result.combine(eigenstate_result)
+        result.computed_lattice_energies = eigenstate_result.eigenenergies
+        return result
+
+    def hopping_qeom_ops(
+        self,
+        qubit_converter: QubitConverter,
+        excitations: Union[
+            str,
+            int,
+            List[int],
+            Callable[[int, Tuple[int, int]], List[Tuple[Tuple[int, ...], Tuple[int, ...]]]],
+        ] = "sd",
+    ) -> None:
```

Comment:
Despite this not being implemented, I would keep the return type hint unchanged for consistency (and also to document future expectation of what this method should return)